src/AdaptedComposition.h

```diff
@@ -57,6 +58,13 @@ class AdaptedCompositionFst : public IComposition {
   // possible optimizations : limit to const FST or limit to StdVectorFst
   const fst::StdFst &fstA_;
```

Comment:
How prevalent is this pattern used in this repo? This opens us to dangling reference bugs. E.g.

```
AdaptedCompositionFst f() {
    fst::StdFst a;
    fst::StdFst b;
    AdaptedCompositionFst c(a, b);
    return c; /* After this is returned a and b have gone out of scope and been de-initialized but c still has references to them */
}
```